src/NuGet.Core/NuGet.Configuration/Settings/ISettings.cs

```diff
@@ -113,15 +108,51 @@ public interface ISettings
         /// <param name="section">The name of the section.</param>
         /// <param name="key">The key to be delted.</param>
         /// <returns>bool indicating success.</returns>
+        [Obsolete("DeleteValue(...) is deprecated, please use Remove(...) with the item you want to remove from the setttings.")]
         bool DeleteValue(string section, string key);
 
         /// <summary>
         /// Deletes the specified <paramref name="section" />.
         /// </summary>
         /// <param name="section">The name of the section.</param>
         /// <returns>bool indicating success.</returns>
+        [Obsolete("DeleteSection(...) is deprecated,, please use Remove(...) with all the items in the section you want to remove from the setttings.")]
         bool DeleteSection(string section);
 
+        /// <summary>
+        /// Gets the section with a given name.
+        /// </summary>
+        /// <param name="sectionName">name to match sections</param>
+        /// <returns>null if no section with the given name was found</returns>
+        SettingSection GetSection(string sectionName);
+
+        /// <summary>
+        /// Adds or updates the given <paramref name="item"/> to the settings.
+        /// If the <paramref name="item"/> has to be added this method will add it
+        /// in the user wide settings file, or walk down the hierarchy (starting from the user wide config)
+        /// until it finds a config where the given section is not cleared.
+        /// </summary>
+        /// <param name="sectionName">section where the <paramref name="item"/> has to be added. If this section does not exist, one will be created.</param>
+        /// <param name="item">item to be added to the settings.</param>
+        /// <returns>true if the item was successfully updated or added in the settings</returns>
+        void AddOrUpdate(string sectionName, SettingItem item);
```

Comment:
change the method return trype to bool

Reply:
I explicitly made this void because I wanted to give more info to the user in the case it didn't succeed. If the add or update failed is because it is trying to update a machine-wide config, therefore I want the user to have this information instead of just returning false and ignoring the changes.